From 3a6b50bb442470565e69fbe2ee00f9e6be2b1958 Mon Sep 17 00:00:00 2001
From: Matthias Klumpp <matthias@tenstral.net>
Date: Mon, 24 Apr 2023 00:27:42 +0200
Subject: [PATCH] Don't crash when non-YAML documents are read as YAML

---
 src/as-distro-extras.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/as-distro-extras.c b/src/as-distro-extras.c
index ed29109e5..3525db747 100644
--- a/src/as-distro-extras.c
+++ b/src/as-distro-extras.c
@@ -130,12 +130,12 @@ as_get_yml_data_origin (const gchar *fname)
 		return NULL;
 	/* start points to the start of the document, i.e. "File:" normally */
 	start = g_strstr_len (data, 400, YAML_SEPARATOR) + YAML_SEPARATOR_LEN;
-	if (start[0] == '\0')
+	if (start == NULL || start[0] == '\0')
 		return NULL;
 	/* Find the end of the first document - can be NULL if there is only one,
 	 * for example if we're given YAML for an empty archive */
 	end = g_strstr_len (start, -1, YAML_SEPARATOR);
-	str = g_strndup (start, strlen(start) - (end ? strlen(end) : 0));
+	str = g_strndup (start, strlen (start) - (end ? strlen (end) : 0));
 
 	strv = g_strsplit (str, "\n", -1);
 	for (i = 0; strv[i] != NULL; i++) {
